pkg/datasource/datasource.go

```diff
@@ -41,7 +45,90 @@ var (
 	ErrOSSNoConfig          = errors.New("no bucket or object config")
 )
 
+type ChunkUploaderConf struct {
```

Comment:
we can add a new file `oss.go` to host the datasource implemetations for Object Storage Service. And move `ChunkUpload`  to `oss.go` as special interafces for `oss`